Invalid content type error fix

bitcloud opened the pull request:
If the user submits an invalid content type, the swagger validator doesn't set the failedValidation field in the error object.

whitlockjc commented:
I'm not sure I agree with the approach taken.  Here are my concerns:

* There are a lot of changes for what could had been just a few lines of change.  I don't understand the need for so much work just to set the `failedValidation` property on the error
* This PR breaks consistency with request validation error messages

If I were fixing this, I would use the same approach [here](https://github.com/apigee-127/swagger-tools/blob/master/middleware/swagger-validator.js#L347) that I do [here](https://github.com/apigee-127/swagger-tools/blob/master/middleware/swagger-validator.js#L232).  That would be a few lines change maximum.

bitcloud commented:
The amount of lines I added are around the same, the rest is just change of the indenting.
True looks like a lot in the diff view. I changed it that way to be consistent with the rest of the error handling in that file. didn't see any reason to handle that differently.

The only thing that changes to the outside is an anyway dynamic error message as far as I know.  (the required and the offending content types are inserted) True, thats the breaking change there. But it should be more consistent in handling (just checking for the failedValidation flag, same error handling in code, same prefix as for the other messages)

catching the error and just adding the flag seems a bit hacky for me as most errors already have that flag ;-)

thought it would be cleaner that way.

whitlockjc commented:
I went ahead and fixed this myself: https://github.com/apigee-127/swagger-tools/commit/beded2129620aa92df0d971508999bd8aa94d42b  I really appreciate your help but I think the simplest approach was to just stick to the existing standard and not to deviate from existing error message patterns.  The end result is the same, the implementation is just a little different.
